deep_quoridor/test/quoridor_test.py

```diff
@@ -0,0 +1,290 @@
+
+    The board should contain N rows for the cells, which are composed by:
+    - '1': Position of player 1
+    - '2': Position of player 2
+    - '.': Empty cell
+    - '*': Potential move positions for player 1
+    - '|': Vertical walls
+    - 'v': A vertical wall cannot be placed here (for testing wall placement)
+    - ' ': Empty spaces (just for formatting)
+    - '+': Vertical wall continuation (just for formatting)
+
+    Additionally, between 2 rows for the cells, a row with horizontal walls can be included.
+    The position of the wall will be with respect to the row above.  The row is composed by:
+    - '-': Horizontal walls
+    - '>': A horizontal wall cannot be placed here (for testing wall placement)
+    - ' ': Empty spaces (just for formatting)
+    - '+': Horizontal wall continuation (just for formatting)
+
+    Returns:
+        tuple: A tuple containing three elements:
+            - Quoridor: A game instance representing the parsed board state
+            - list: A list of tuples (row, col) representing potential move positions
+            - list: A list of tuples (row, col, orientation) representing where the walls are forbidden to be placed
+
+    """
+    rows = [r for r in board.split("\n") if r.strip()]
+    size = len([ch for ch in rows[0] if ch in "12.*"])
+    board = Board(size, 10)
+    potential_moves = []
+    forbidden_walls = []
+
+    row_n = 0
+    col_positions = {}  # map from position in the string to column number
+
+    for row in rows:
+        col_n = 0
+
+        is_cell_row = set(row) <= set("12.*| +v")
+        is_h_wall_row = set(row) <= set("- +>")
+        assert is_cell_row or is_h_wall_row, f"Row contains invalid characters {row}"
+
+        if is_h_wall_row:
+            for i, col_n in col_positions.items():
+                if i >= len(row):
+                    break
+
+                ch = row[i]
+                if ch == "-":
+                    # HACK: Since each wall has two "|" characters, this tries to place each wall twice, and
+                    # the second time is off by one position. Checking whether we can place it stops us
+                    # placing the second wall.
+                    if board.can_place_wall((row_n - 1, col_n), WallOrientation.HORIZONTAL):
+                        board.add_wall(0, (row_n - 1, col_n), WallOrientation.HORIZONTAL)
+
+                if ch == ">" or ch == "-":
+                    forbidden_walls.append((row_n - 1, col_n, WallOrientation.HORIZONTAL))
+
+            col_positions = {}
+
+        else:
+            for i, ch in enumerate(row):
+                if ch == " " or ch == "+":
+                    continue
+
+                if ch == "v" or ch == "|":
+                    forbidden_walls.append((row_n, col_n - 1, WallOrientation.VERTICAL))
+                    if ch == "|":
+                        # We arbitrarily use player 1 to place the walls since it doesn't matter for these tests.
```

Comment:
Maybe we should be placing walls alternating between player 1 and player 2, since now it does matter (e.g. I think there's a bug because it just checks walls for player 1)

Reply:
Great point.